backend/common/src/main/java/ai/verta/modeldb/common/event/FutureEventDAO.java

```diff
@@ -0,0 +1,111 @@
+
+  public FutureEventDAO(Executor executor, FutureJdbi jdbi, Config config, String serviceType) {
+    this.executor = executor;
+    this.jdbi = jdbi;
+    this.config = config;
+    this.serviceType = serviceType;
+  }
+
+  public Boolean addLocalEventWithBlocking(
+      String resourceType, String eventType, long workspaceId, JsonObject eventMetadata) {
+    return addLocalEvent(resourceType, eventType, workspaceId, eventMetadata).get();
+  }
+
+  public InternalFuture<Boolean> addLocalEventWithAsync(
+      String resourceType, String eventType, long workspaceId, JsonObject eventMetadata) {
+    return addLocalEvent(resourceType, eventType, workspaceId, eventMetadata);
+  }
+
+  private InternalFuture<Boolean> addLocalEvent(
+      String resourceType, String eventType, long workspaceId, JsonObject eventMetadata) {
+    if (!config.isEvent_system_enabled()) {
+      LOGGER.info("Event system is not enabled");
+      return InternalFuture.completedInternalFuture(false);
+    }
+
+    if (workspaceId == 0) {
+      LOGGER.info("workspaceId not found while logging event");
+      return InternalFuture.failedStage(new ModelDBException("Workspace id should not be empty"));
+    }
+
+    eventMetadata.addProperty("service", serviceType);
+    eventMetadata.addProperty("resource_type", resourceType);
+    eventMetadata.addProperty("logged_time", new Date().getTime());
+
+    return jdbi.withHandle(
+            handle ->
+                handle
+                    .createUpdate(
+                        "insert into event (event_uuid, event_type, workspace_id, event_metadata) values (:event_uuid, :event_type, :workspace_id, :event_metadata) ")
+                    .bind("event_uuid", UUID.randomUUID().toString())
+                    .bind("event_type", eventType)
+                    .bind("workspace_id", workspaceId)
+                    .bind("event_metadata", eventMetadata.toString())
+                    .execute())
+        .thenApply(
+            insertedRowCount -> {
+              LOGGER.debug("Event added successfully");
+              return insertedRowCount > 0;
+            },
+            executor);
+  }
+
+  public Boolean deleteLocalEventWithBlocking(List<String> eventUUIDs) {
+    return deleteLocalEvent(eventUUIDs).get();
+  }
+
+  public InternalFuture<Boolean> deleteLocalEventWithAsync(List<String> eventUUIDs) {
+    return deleteLocalEvent(eventUUIDs);
+  }
+
+  private InternalFuture<Boolean> deleteLocalEvent(List<String> eventUUIDList) {
+    if (!config.isEvent_system_enabled()) {
+      LOGGER.info("Event system is not enabled");
+      return InternalFuture.completedInternalFuture(false);
+    }
+
+    Set<String> eventUUIDs =
+        eventUUIDList.stream()
+            .filter(eventUUID -> !eventUUID.isEmpty())
+            .collect(Collectors.toSet());
+
+    if (eventUUIDs.isEmpty()) {
+      return InternalFuture.failedStage(new ModelDBException("Event UUID not found for deletion"));
```

Comment:
Should we consider deleting no events to be a failure or a success? I would consider deleting no events when no events were provided to be a success, but would specify "0 out of 0 events deleted" or something.